alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt

```diff
@@ -372,6 +381,42 @@ class TransportIndexWorkflowAction @Inject constructor(
                     )
                     return
                 }
+
+                val createdWorkflow = request.workflow.copy(id = indexResponse.id)
+                val executionId = CompositeWorkflowRunner.generateExecutionId(false, createdWorkflow)
+
+                val (workflowMetadata, _) = WorkflowMetadataService.getOrCreateWorkflowMetadata(
+                    workflow = createdWorkflow,
+                    skipIndex = false,
+                    executionId = executionId
+                )
+
+                val delegates = (createdWorkflow.inputs[0] as CompositeInput).sequence.delegates.sortedBy { it.order }
+                val monitors = monitorCtx.workflowService!!.getMonitorsById(delegates.map { it.monitorId }, delegates.size)
+
+                for (monitor in monitors) {
+                    val (monitorMetadata, created) = MonitorMetadataService.getOrCreateMetadata(
+                        monitor = monitor,
+                        createWithRunContext = true,
+                        workflowMetadataId = workflowMetadata.id
+                    )
+
+                    if (created == false) {
+                        log.warn("Metadata doc id:${monitorMetadata.id} exists, but it shouldn't!")
+                    }
+
+                    if (monitor.monitorType == Monitor.MonitorType.DOC_LEVEL_MONITOR) {
+                        docLevelMonitorQueries.indexDocLevelMonitorQueries(
+                            monitor,
+                            monitor.id,
+                            monitorMetadata,
+                            request.refreshPolicy,
+                            indexTimeout
+                        )
+                    }
```

Comment:
Why is this needed? This happens when the monitor is first created anyways.

Reply:
fixed it.